includes/handlers/class-webmention-handler-meta.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * Class for webmention parsing using META tags.
+*/
+class Webmention_Handler_Meta extends Webmention_Handler_Base {
+
+	/**
+	 * Takes a request object and parses it.
+	 *
+	 * @param Webmention_Request $request Request Object.
+	 * @param Webmention_Item $item A Parsed Item. If null, a new one will be created.
+	 * @return WP_Error|true Return error or true if successful.
+	 */
+	public function parse( $request, $item = null ) {
+		if ( $item instanceof Webmention_Item ) {
+			$this->webmention_item = $item;
+		}
+		$dom   = clone $request->get_domdocument();
+		$xpath = new DOMXPath( $dom );
+
+		$meta = array();
+		// Look for OGP properties
+		foreach ( $xpath->query( '//meta[(@name or @property or @itemprop) and @content]' ) as $tag ) {
+			$meta_name = $tag->getAttribute( 'property' );
+			if ( ! $meta_name ) {
+				$meta_name = $tag->getAttribute( 'name' );
+			}
+			if ( ! $meta_name ) {
+				$meta_name = $tag->getAttribute( 'itemprop' );
+			}
+			$meta_value = $tag->getAttribute( 'content' );
+			// Sanity check. $key is usually things like 'title', 'description', 'keywords', etc.
+			if ( strlen( $meta_name ) > 200 ) {
+				continue;
+			}
+			$meta = self::add_property( $meta, $meta_name, $meta_value );
+		}
+
+		$meta['title'] = trim( $xpath->query( '//title' )->item( 0 )->textContent );
+		$meta          = self::parse_ogp( $meta );
+		if ( isset( $meta['og'] ) ) {
+			$meta['og'] = self::parse_ogp( $meta['og'] );
+		}
+
+		$this->webmention_item = new Webmention_Item();
```

Comment:
this replaces the item if there is one or not. I would like to require the item as param.